packages/ui/src/App/index.tsx

```diff
@@ -30,119 +30,48 @@ socket.addEventListener("message", event => {
   }
 });
 
-const ComponentPageWithProps = (
-  props: {context: Context} & RouteComponentProps<{
-    path: string;
-  }>,
-) => {
-  const matchingPath = ({path}: {path: string}) =>
-    path === props.match.params.path;
-  const components = props.context.components.filter(matchingPath);
-  const fileError = props.context.errors.filter(matchingPath);
-  const errors = fileError.length !== 0 ? fileError[0].errors : [];
-
-  if (components.length === 0) {
-    return (
-      <NotFoundPage
-        filepath={props.match.params.path}
-        context={props.context}
-      />
-    );
-  }
-
-  return <ComponentPage component={components[0]} errors={errors} {...props} />;
-};
-
-const WithContext = <Props extends {}>(
+const WithContextProvider = <Props extends {}>(
   ComponentNeedingContext: React.ComponentType<Props>,
-  selectedPaths?: Set<FilePath>,
-) => (props: Props) => {
-  return (
-    <ComponentNeedingContext
-      context={context}
-      selectedPaths={selectedPaths}
-      {...props}
+) => (props: Props) => (
+  <Provider store={store}>
+    <ComponentNeedingContext {...props} />
+  </Provider>
```

Comment:
Can we rename this function - doesn't really make sense any more.

Reply:
In fact, just wrap the whole app in a `Provider`

Reply:
The whole app is wrapped. I just pulled out the wrapping into a HOC. I can remove it as it's not really necessary.